src/box/gc.c

```diff
@@ -633,6 +655,7 @@ gc_ref_checkpoint(struct gc_checkpoint *checkpoint,
 void
 gc_unref_checkpoint(struct gc_checkpoint_ref *ref)
 {
+	assert(gc.cleanup_fiber != NULL);
```

Comment:
Come to think of it, I don't like how these assertions look (what `cleanup_fiber` has to do with `gc_unref_checkpoint`?). Please remove. Let's rely on the fact that `fiber_wakeup` should crash if the fiber was joined.

Reply:
Done.

<details>
<summary>diff</summary>

```diff
--- a/src/box/gc.c
+++ b/src/box/gc.c
@@ -322,7 +322,6 @@ gc_cleanup_fiber_f(va_list ap)
 void
 gc_set_wal_cleanup_delay(double wal_cleanup_delay)
 {
-	assert(gc.cleanup_fiber != NULL);
 	gc.wal_cleanup_delay = wal_cleanup_delay;
 	/*
 	 * This routine may be called at arbitrary
@@ -346,7 +345,6 @@ gc_delay_ref(void)
 void
 gc_delay_unref(void)
 {
-	assert(gc.cleanup_fiber != NULL);
 	if (gc.is_paused) {
 		assert(gc.delay_ref > 0);
 		gc.delay_ref--;
@@ -390,7 +388,6 @@ gc_wait_cleanup(void)
 void
 gc_advance(const struct vclock *vclock)
 {
-	assert(gc.cleanup_fiber != NULL);
 	/*
 	 * In case of emergency ENOSPC, the WAL thread may delete
 	 * WAL files needed to restore from backup checkpoints,
@@ -434,7 +431,6 @@ gc_set_min_checkpoint_count(int min_checkpoint_count)
 void
 gc_set_checkpoint_interval(double interval)
 {
-	assert(gc.checkpoint_fiber != NULL);
 	/*
 	 * Reconfigure the schedule and wake up the checkpoint
 	 * daemon so that it can readjust.
@@ -452,7 +448,6 @@ gc_set_checkpoint_interval(double interval)
 void
 gc_add_checkpoint(const struct vclock *vclock)
 {
-	assert(gc.cleanup_fiber != NULL);
 	struct gc_checkpoint *last_checkpoint = gc_last_checkpoint();
 	if (last_checkpoint != NULL &&
 	    vclock_sum(&last_checkpoint->vclock) == vclock_sum(vclock)) {
@@ -543,7 +538,6 @@ gc_do_checkpoint(bool is_scheduled)
 int
 gc_checkpoint(void)
 {
-	assert(gc.checkpoint_fiber != NULL);
 	if (gc.checkpoint_is_in_progress) {
 		diag_set(ClientError, ER_CHECKPOINT_IN_PROGRESS);
 		return -1;
@@ -583,7 +577,6 @@ gc_checkpoint(void)
 void
 gc_trigger_checkpoint(void)
 {
-	assert(gc.checkpoint_fiber != NULL);
 	if (gc.checkpoint_is_in_progress || gc.checkpoint_is_pending)
 		return;
 
@@ -655,7 +648,6 @@ gc_ref_checkpoint(struct gc_checkpoint *checkpoint,
 void
 gc_unref_checkpoint(struct gc_checkpoint_ref *ref)
 {
-	assert(gc.cleanup_fiber != NULL);
 	rlist_del_entry(ref, in_refs);
 	gc_schedule_cleanup();
 }
@@ -683,7 +675,6 @@ gc_consumer_register(const struct vclock *vclock, const char *format, ...)
 void
 gc_consumer_unregister(struct gc_consumer *consumer)
 {
-	assert(gc.cleanup_fiber != NULL);
 	if (!consumer->is_inactive) {
 		gc_tree_remove(&gc.consumers, consumer);
 		gc_schedule_cleanup();
@@ -694,7 +685,6 @@ gc_consumer_unregister(struct gc_consumer *consumer)
 void
 gc_consumer_advance(struct gc_consumer *consumer, const struct vclock *vclock)
 {
-	assert(gc.cleanup_fiber != NULL);
 	if (consumer->is_inactive)
 		return;
 

```

</details>